python/rmm/_lib/memory_resource.pyx

```diff
@@ -12,16 +13,37 @@ from libcpp.string cimport string
 from rmm._cuda.gpu import CUDARuntimeError, cudaError, getDevice, setDevice
 
 
-cdef class CudaMemoryResource(MemoryResource):
-    def __cinit__(self, device=None):
-        if device is None:
-            self.c_obj.reset(
-                new cuda_memory_resource_wrapper()
-            )
+cdef class DeviceMemoryResource:
+
+    cdef device_memory_resource* get_mr(self):
+        return self.c_obj.get()
+
+
+cdef class UpstreamResourceAdaptor(DeviceMemoryResource):
+
+    def __cinit__(self, *args, **kwargs):
```

Comment:
Can we make this signature something like the following:

```suggestion
def __cinit__(self, DeviceMemoryResource upstream_mr, *args, **kwargs)
```

if that works, we might be able to avoid args/kwargs introspection.

Reply:
Good idea. I'll try it out and let you know.

Reply:
This worked well. Great suggestion.